src/software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_test.py

```diff
@@ -0,0 +1,167 @@
+                .enemyDefenseArea()
+                .negXNegYCorner(),
+                tbots_cpp.Field.createSSLDivisionBField()
+                .enemyDefenseArea()
+                .negXPosYCorner(),
+            ],
+            tbots_cpp.Point(-2, 2),
+        ),
+        (
+            [
+                tbots_cpp.Point(-2.75, 2.5),
+                tbots_cpp.Point(-2.75, 1.5),
+                tbots_cpp.Point(-2.75, 0.5),
+                tbots_cpp.Point(-2.75, -0.5),
+                tbots_cpp.Point(-2.75, -1.5),
+                tbots_cpp.Point(-2.75, -3.0),
+            ],
+            [
+                tbots_cpp.Point(1.8, -2),
+                tbots_cpp.Point(0, -2.5),
+                tbots_cpp.Point(2, 0),
+                tbots_cpp.Field.createSSLDivisionBField().enemyGoalCenter(),
+                tbots_cpp.Field.createSSLDivisionBField()
+                .enemyDefenseArea()
+                .negXNegYCorner(),
+                tbots_cpp.Field.createSSLDivisionBField()
+                .enemyDefenseArea()
+                .negXPosYCorner(),
+            ],
+            tbots_cpp.Point(2, -2),
+        ),
+    ],
+)
+def test_enemy_free_kick_play(
+    simulated_test_runner, blue_bots, yellow_bots, ball_initial_pos
+):
+    # Setup Bots
+    def setup(*args):
+
+        # Game Controller Setup
+        simulated_test_runner.gamecontroller.send_gc_command(
+            gc_command=Command.Type.STOP, team=Team.UNKNOWN
+        )
+        simulated_test_runner.gamecontroller.send_gc_command(
+            gc_command=Command.Type.DIRECT, team=Team.YELLOW
+        )
+
+        # Force play override here
+        blue_play = Play()
+        blue_play.name = PlayName.EnemyFreeKickPlay
+
+        yellow_play = Play()
+        yellow_play.name = PlayName.FreeKickPlay
+
+        simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+        simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+            Play, yellow_play
+        )
+
+        # Create world state
+        simulated_test_runner.simulator_proto_unix_io.send_proto(
+            WorldState,
+            create_world_state(
+                yellow_robot_locations=yellow_bots,
+                blue_robot_locations=blue_bots,
+                ball_location=ball_initial_pos,
+                ball_velocity=tbots_cpp.Vector(0, 0),
+            ),
+        )
+
+    # Always Validation
+    always_validation_sequence_set = [
+        [
+            OrValidation(
+                [
+                    RobotNeverEntersRegion(
+                        regions=[tbots_cpp.Circle(ball_initial_pos, 0.05)]
+                    ),
+                    BallEventuallyMovesFromRest(position=ball_initial_pos),
+                ]
+            )
+        ]
+    ]
+    # Eventually Validation
+    eventually_validation_sequence_set = [
+        [RobotEventuallyEntersRegion(regions=[tbots_cpp.Circle(ball_initial_pos, 1)])]
```

Comment:
why is this validation important for this test?

I don't mind keeping it in but is there a specific reason why you wanted it?

Reply:
Oh yeah, this is to ensure that there is at least one defender assigned to the ball (since we don't want to leave the kicker unguarded).

Reply:
makes sense, thanks